dc_make/maker.py

```diff
@@ -241,6 +245,24 @@ def writeDatacards(self, output):
       shape_file = os.path.join(output, f"{proc_name}.root")
       self.cb.cp().mass(param_list).process(processes).WriteDatacard(dc_file, shape_file)
 
+      for subera in self.eras:
+        for subchannel in self.channels:
+          tmp_output = output+f'/{subera}/{subchannel}/'
+          os.makedirs(tmp_output, exist_ok=True)
+          tmp_dc_file = os.path.join(tmp_output, f"datacard_{proc_name}.txt")
+          tmp_shape_file = os.path.join(tmp_output, f"{proc_name}.root")
+          self.cb.cp().era([subera]).channel([subchannel]).mass(param_list).process(processes).WriteDatacard(tmp_dc_file, tmp_shape_file)
```

Comment:
In this way, root files are copied multiple times. You should always the same file and use create the combined version at the end (to have all histograms stored)

Reply:
Fixed, now creates the total shape at the end and overwrites the temporary ones